src/components/ec-table-sort/ec-table-sort.vue

```diff
@@ -23,32 +23,28 @@
   </a>
 </template>
 
-<script setup>
+<script setup lang="ts">
 import { computed } from 'vue';
 
-import * as SortDirection from '../../enums/sort-direction';
+import { SortDirection } from '../../enums/sort-direction';
 import EcIcon from '../ec-icon';
+import { IconName } from '../ec-icon/types';
+import type { TableSortEvent, TableSortEvents, TableSortProps } from './types';
 
-const emit = defineEmits(['sort']);
+const emit = defineEmits<{
+  'sort': [value: TableSortEvents[TableSortEvent.SORT]],
+}>();
 
-const props = defineProps({
-  direction: {
-    type: String,
-    default: null,
-    validator(value) {
-      return value === null || value === '' || [SortDirection.ASC, SortDirection.DESC].includes(value);
-    },
-  },
-});
+const props = defineProps<TableSortProps>();
```

Comment:
we had a default value for `direction`, why it's been removed?

Reply:
My bad, it should not have been removed, I'm adding it